sqrl-engines/sqrl-engine-flink/src/main/java/com/datasqrl/engine/stream/flink/plan/SqrlToFlinkExecutablePlan.java

```diff
@@ -195,30 +198,36 @@ private static RexNode convertField(RelDataTypeField field, AtomicBoolean hasCha
     boolean hasNativeSupport = field.getType() instanceof RawRelDataType && engine.isPresent() &&
         engine.get().supportsType(((RawRelDataType) field.getType()).getRawType().getDefaultConversion());
 
-    Optional<SqlOperator> downcastFunc = Optional.empty();
     if (field.getType() instanceof RawRelDataType && !hasNativeSupport) {
       Class<?> defaultConversion = ((RawRelDataType) field.getType()).getRawType()
           .getDefaultConversion();
       CatalogReader catalogReader = (CatalogReader) relBuilder.getRelOptSchema();
+
       DowncastFunction downcastFunction = ServiceLoaderDiscovery.get(DowncastFunction.class,
           e -> e.getConversionClass().getName(),
           defaultConversion.getName());
-      downcastFunc = catalogReader.getOperatorList().stream()
-          .filter(f -> f.getName().equalsIgnoreCase(downcastFunction.downcastFunctionName()))
-          .findFirst();
-      if (downcastFunc.isPresent()) {
-        downcastClassNames.put(downcastFunc.get().getName(), downcastFunction.getDowncastClassName()
-            .getName());
+      if (downcastFunction == null) {
+        throw new RuntimeException("Needed downcast function but could not find one: " + defaultConversion.getName());
       }
-    }
+      String fncName = downcastFunction.downcastFunctionName().toLowerCase();
+
+      List<SqlOperator> operators = new ArrayList<>();
+      catalogReader.getSchema().getSqrlFramework()
+          .getSqrlOperatorTable().lookupOperatorOverloads(
```

Comment:
This is still incorrect, the downcast function may not have been imported. Probably a reflection construction is best here since we'll need the return type for proper inference.